### PR TITLE
feat: Auth client setup

### PR DESCRIPTION
**Pull request by adilansari**

### CODE REVIEW

File: lib/tigris.ts [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
+const tigrisUri: string = process.env.TIGRIS_URI as string
 
 declare global {
   var tigrisClient: Tigris
 }
 
+let tigris: Tigris
+let clientConfig: TigrisClientConfig
+
+if (process.env.NODE_ENV === 'production' || requiresAuth(tigrisUri)) {
```
Comment (ovaistariq) [inline lib/tigris.ts:16]: Shouldn't we force auth as a best practice if env is production?

File: .gitignore [hunk 2]
```diff
@@ -69,6 +69,8 @@ typings/
 .yarn-integrity
 
 # dotenv environment variables file
+.env.development.local
```
Comment (ovaistariq) [inline .gitignore:72]: do you want to have a `.local` postfix in these files?

File: lib/tigris.ts (hunk 1 above)
Comment (adilansari) [reply lib/tigris.ts:16]: This is already doing that, and additionally enforcing auth for dev when uri is not localhost etc (tigris cloud from dev).

File: .gitignore (hunk 2 above)
Comment (adilansari) [reply .gitignore:72]: This is to only prevent accidental commit of dev/prod credentials. My understanding is that [these](https://nextjs.org/docs/basic-features/environment-variables#environment-variable-load-order) are all valid env files for nextjs. (I will replace these with env.development* and end.prod*)

File: lib/tigris.ts (hunk 1 above)
Comment (ovaistariq) [reply lib/tigris.ts:16]: Right but this could be simplified further by just keep the environment check.

**Pull request merged**